AAE-395 Support extensions for all model types

pmartinezga opened the pull request.

pmartinezga committed:
ACTIVITI-3885 Added new metadata attribute to models

Added new attribute to models to store additional metadata when needed and include support for validating that metadata depending on the model type.

pmartinezga committed:
AAE-395 Added new metadata attribute to models

Added new attribute to models to store additional metadata when needed and include support for validating that metadata depending on the model type.

pmartinezga committed:
ACTIVITI-3885 Added new metadata attribute to models

Added new attribute to models to store additional metadata when needed and include support for validating that metadata depending on the model type.

pmartinezga committed:
ACTIVITI-3885 Add extensions support for any model

Extensions is now a map of string, object so it fits any object
Added validation mechanism for content and extensions
Added a content update event where listeners can register
Add new API endpoint to validate extensions files

pmartinezga committed:
AAE-395 Add extensions support for any model

Extensions is now a map of string, object so it fits any object
Added validation mechanism for content and extensions
Added a content update event where listeners can register
Add new API endpoint to validate extensions files

pmartinezga committed:
ACTIVITI-3885 Add extensions support for any model

Extensions is now a map of string, object so it fits any object
Added validation mechanism for content and extensions
Added a content update event where listeners can register
Add new API endpoint to validate extensions files

pmartinezga committed:
ACTIVITI-3885 Code refactoring

pmartinezga committed:
AAE-395 Code refactoring

pmartinezga committed:
ACTIVITI-3885 Code refactoring

pmartinezga committed:
AAE-395 Fix acceptance tests

pmartinezga committed:
ACTIVITI-3885 Fix acceptance tests

pmartinezga committed:
ACTIVITI-3885 Fix acceptance tests

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/Activiti/activiti-cloud-org-service?pullRequest=363) <br/>All committers have signed the CLA.

pmartinezga committed:
AAE-395 Rebasing develop

ffazzini commented on activiti-cloud-services-org/activiti-cloud-services-org-rest/src/test/java/org/activiti/cloud/services/organization/config/GenericModelsConfiguration.java:
```diff
@@ -0,0 +1,186 @@
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+
+@Profile("generic")
```
why the profile ?

ffazzini commented on activiti-cloud-services-org/activiti-cloud-services-org-service/src/main/java/org/activiti/cloud/services/organization/validation/extensions/ProcessExtensionsModelValidator.java:
```diff
@@ -0,0 +1,139 @@
+    @Override
+    protected Stream<ModelValidationError> validateModelExtensions(Model model,
+                                                                   ValidationContext context) {
+        return Optional.ofNullable(model.getId())
```
would not without the stream make the code easier to understand in this specific case? Why returning a stream if we expect a limited number of results?

pmartinezga replied:
Ok, changed to list and renamed so it is easier to understand the code

pmartinezga committed:
AAE-395 Changed extensions model validators

Changed the method return type of the methods that retireve the validation errors from Stream to List.
Also rename the methods so it fits better to what they do

pmartinezga committed:
AAE-395 Merge develop

pmartinezga committed:
AAE-395 Changed extensions model validators

Changed the method return type of the methods that retireve the validation errors from Stream to List.
Also rename the methods so it fits better to what they do

pmartinezga committed:
AAE-395 Rebasing develop

pmartinezga committed:
AAE-395 Fix id issues when importing projects

ffazzini commented on activiti-cloud-services-org/activiti-cloud-services-org-rest/src/test/java/org/activiti/cloud/services/organization/rest/controller/GenericJsonModelTypeContentUpdateListenerControllerIT.java:
```diff
@@ -0,0 +1,116 @@
+    }
+
+    @Test
+    public void testUpdateContentGenericJsonModelCallsContentUpdateListenerForTheModel() throws Exception {
```
new test naming convention in all tests

ffazzini replied:
we also decided to remove comments //GIVEN //WHEN //THEN

pmartinezga replied:
Renamed all the test cases and removed "given-when-then" comments

ffazzini commented on activiti-cloud-services-org/activiti-cloud-services-org-service/src/main/java/org/activiti/cloud/services/organization/validation/extensions/ExtensionsModelValidator.java:
```diff
@@ -1,171 +1,50 @@
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
+ * Copyright 2005-2018 Alfresco Software, Ltd. All rights reserved.
```
restore community header, please check if that happened in other files too

pmartinezga replied:
Restored, and checked that it didn't happen anywhere else

pmartinezga committed:
AAE-395 Rename tests

pmartinezga committed:
AAE-395 Merge develop

mergify[bot] committed:
Merge branch 'develop' into dev-pmartinezga-ACTIVITI-3885
